channeldb/meta.go

```diff
@@ -0,0 +1,78 @@
+package channeldb
+
+import (
+	"github.com/boltdb/bolt"
+)
+
+var (
+	// metaBucket stores all the meta information concerning the state of
+	// the database.
+	metaBucket = []byte("metadata")
+
+	// dbVersionKey is a boltdb key and it's used for storing/retrieveing
+	// current database version.
+	dbVersionKey = []byte("dbp")
+)
+
+// Meta structure holds the database meta information.
+type Meta struct {
+	dbVersionNumber uint32
+}
+
+func (d *DB) FetchMeta() (*Meta, error) {
+	meta := &Meta{}
+	err := d.store.View(func(tx *bolt.Tx) error {
+		metaBucket := tx.Bucket(metaBucket)
+		if metaBucket == nil {
+			return ErrMetaNotFound
+		}
+
+		fetchDbVersion(metaBucket, meta)
+
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return meta, nil
+}
+
+func (d *DB) PutMeta(meta *Meta) error {
+	err := d.store.Update(func(tx *bolt.Tx) error {
```

Comment:
Since no further processing in the function takes place past updating the meta-data state, one can simply do: 
```go
return d.store.Update(func(tx *bolt.Tx) error {
      .... 
})
```